padl/transforms.py

```diff
@@ -2248,3 +2301,779 @@ def importdump(transform_or_module):
+                continue
+            path_copy = path.copy()
+            if isinstance(out_node.transform, Graph):
+                path_copy.append(out_node.transform.input_node)
+                return_path.append(out_node.transform.list_all_paths(out_node.transform.input_node, path_copy))
+            else:
+                path_copy.append(out_node)
+                return_path.append(self.list_all_paths(out_node, path_copy))
+        if len(return_path) == 1:
+            return return_path[0]
+        return return_path
+
+    def _generate_preprocess_dict(self,
+                                  current_node,
+                                  preprocess_edges=None,
+                                  batchify_node=None,
+                                  batchify_found=False):
+        """
+
+        :param current_node:
+        :param preprocess_edges:
+        :param batchify_node:
+        :param batchify_found:
+        :return:
+        """
+        if batchify_node is None:
+            batchify_node = dict()
+        if preprocess_edges is None:
+            preprocess_edges = defaultdict(dict)
+
+        # import pdb;pdb.set_trace()
+        for child in self.edges[current_node]:
+            # import pdb;pdb.set_trace()
+            if child.transform == padl.batch:
+                batchify_found = True
+                if child not in batchify_node:
+                    batchify_node[child] = Node(padl.Identity() - 'batchify Marker')
+                preprocess_edges[current_node][batchify_node[child]] = self.edges[current_node][child]
+                preprocess_edges[batchify_node[child]][self.output_node] = (None, None)
+            elif child == self.output_node and batchify_found:
+                raise SyntaxError('If a path has batchify, all paths must contain batchify')
+            else:
+                preprocess_edges[current_node][child] = self.edges[current_node][child]
+                preprocess_edges, batchify_found = self._generate_preprocess_dict(child,
+                                                                                  preprocess_edges,
+                                                                                  batchify_node,
+                                                                                  batchify_found)
+        return preprocess_edges, batchify_found
+
+    def _generate_forward_dict(self,
+                               current_node=None,
+                               forward_edges=None,
+                               unbatchify_node=None,
+                               unbatchify_found=False,
+                               ):
+        if unbatchify_node is None:
+            unbatchify_node = dict()
+        if forward_edges is None:
+            forward_edges = defaultdict(dict)
+        if current_node is None:
+            batchify_nodes = [n_ for n_ in self.edges if n_.transform == padl.batch]
```

Comment:
Consider changing to `if isinstance(n_.transform, Batchify)` because the builtin `batch` may not be used.